ingest/scripts/transform-new-fields

```diff
@@ -0,0 +1,50 @@
+#! /usr/bin/env python3
+
+import argparse
+import json
+import csv
+from sys import stdin, stdout
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Transform JSON data by applying a TSV mapping and adding new columns."
+    )
+    parser.add_argument("--map-tsv", required=True,
+        help="Path to the TSV mapping file.")
+    parser.add_argument("--map-id", required=True,
+        help="Column name in the map TSV to use as the mapping key.")
+    parser.add_argument("--metadata-id", required=True,
+        help="Column name in the metadata JSON to use as the mapping key.")
+    parser.add_argument("--map-fields", nargs="+", required=True,
+        help="Columns to add from the mapping file.")
+    parser.add_argument("--pass-through", default=False,
+        help="If set, pass through the original value when no mapping is found.")
```

Comment:
What's the reasoning behind `default=False`? I see it's set to true in all usages ([dengue](https://github.com/nextstrain/dengue/blob/f6ab41aaf10e6002d857fa3fc5912e26eea2eb9f/ingest/rules/curate.smk#L90) and [WNV](https://github.com/nextstrain/WNV/blob/34414e507434d080d15594198a76b70b0a8ba2fe/ingest/rules/curate.smk#L82)). I'd think that the two behaviors would be either pass through or error, not pass through or drop.